app/jobs/place_holds_job.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class PlaceHoldsJob < ApplicationJob
+  queue_as :default
+
+  def perform(patron_key:, barcodes:, session_token:, **hold_args)
+    @barcodes = barcodes
+    @session_token = session_token
+    @hold_args = hold_args
+    @patron = fetch_patron patron_key
+
+    write_to_redis print_html(batch_results)
+  end
+
+  private
+
+    def write_to_redis(html)
+      Redis.current.set("place_holds_results_#{@patron.key}", {
+        id: @patron.barcode.to_s,
+        result: :success,
+        response: html
+      }.to_json)
+    end
+
+    def print_html(results)
+      results_builder = PlaceHoldResults::Builder.new(user_token: @session_token,
+                                                      client: symphony_client,
+                                                      place_hold_results: results)
+
+      ApplicationController.render partial: 'holds/results',
+                                   layout: false,
+                                   locals: { place_hold_results: results_builder.generate }
+    end
+
+    def fetch_patron(patron_key)
+      patron_response = symphony_client.patron_info patron_key: patron_key,
+                                                    session_token: @session_token
+      Patron.new patron_response
+    end
+
+    def batch_results
+      @barcodes.each_with_object(success: [], error: []) do |barcode, status|
+        response = symphony_client.place_hold(@patron, @session_token, barcode, @hold_args)
+
+        sirsi_response = SirsiResponse.new response
+
+        if response.status == 200
+          status[:success] << { barcode: barcode,
+                                hold_key: sirsi_response.hold.key }
+        else
+          Sidekiq.logger.error("place_holds_results_#{@patron_key} #{barcode}: #{sirsi_response.response_raw}")
+          status[:error] << { barcode: barcode,
+                              result: :failure,
```

Comment:
do we need `result: :failure,` here? Sorry, asking to remember after giving a week long break 😬

Reply:
In cases of failure, the polling.js script looks for this key: https://github.com/psu-libraries/myaccount/pull/213/files/b3e2a69a16f07f66d661d72b27fc63a4a36f516f#diff-1f1fb8e7901b4fddb481654b8038b693L33

Reply:
Yeah, last you looked at this class it was basically a big long list of instructions, hopefully a bit easier to read with stuff broken down into methods.

Reply:
ahh thanks, remembered now 👍